extensions-contrib/kafka-segment-metadata-emitter/src/main/java/org/apache/druid/emitter/kafkasegmentmetadata/DruidSegmentEventProto.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.kafkasegmentmetadata;
+
+import io.confluent.streaming.aggregates.proto.DruidSegmentEvent;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.util.Map;
+
+public class DruidSegmentEventProto implements Serde<DruidSegmentEvent>, Serializer<DruidSegmentEvent>, Deserializer<DruidSegmentEvent>
```

Comment:
shall we call this DruidSegmentEventSerde?

Reply:
actually...looks like we just need a serializer 
```
rops.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, DruidSegmentEventProto.class.getName());
```
If so, we can simplify this class to be `DruidSegmentEventSeriazlier implements Serializer<DruidSegmentEvent>`